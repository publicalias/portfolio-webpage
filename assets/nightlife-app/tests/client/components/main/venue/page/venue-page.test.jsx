"use strict";

//local imports

const VenuePage = require("../../../../../../scripts/client/components/main/venue/page/venue-page");

const { newUserWithData } = require("../../../../../../schemas");
const { getGeoPoint, testWrapper } = require("../../../../test-helpers");

//global imports

const { initTestSnapshot, withDataList } = require("redux/tests/client-tests");
const { testMock } = require("redux/tests/meta-tests");
const { reactTests } = require("redux/tests/react-tests");

//utilities

const { testMount, testShallow } = testWrapper(VenuePage);

//setup

beforeAll(reactTests.setup);
beforeEach(reactTests.inject(VenuePage, { lib: { getLocation: jest.fn((user) => user.data.location) } }));

//venue page

describe("VenuePage (general)", () => {

  const dataList = [null, { id: "id-a" }];

  const testSnapshot = withDataList(initTestSnapshot(testShallow), dataList);

  const testPage = withDataList(testMount, dataList);

  const testLoadA = (fn) => {

    const { props, setProps, wrapper } = testPage();

    wrapper.mount();

    setProps(null, { id: "id-b" });

    fn(props);

    wrapper.unmount();

  };

  const testLoadB = async (fn) => {

    const userData = {
      a: { data: { location: null } },
      b: { data: { location: getGeoPoint(0) } }
    };

    const list = [
      [{ user: userData.a }],
      [{ ready: true }],
      [{ notifications: { friends: [{}] } }],
      [{ notifications: { rsvps: [{}] } }],
      [null, { id: "id-b" }],
      [{ user: userData.b }]
    ];

    const { props, setProps, wrapper } = testPage();

    wrapper.mount();

    for (const e of list) {
      setProps(...e);
    }

    await Promise.resolve();

    fn(props, userData);

    wrapper.unmount();

  };

  it("should match snapshot (default)", () => testSnapshot());

  it("should match snapshot (authenticated)", () => testSnapshot({ user: newUserWithData() }));

  it("should match snapshot (match)", () => testSnapshot({ venues: { data: [{ id: "id-a" }] } }));

  it("should call venueClearState conditionally on update", () => testLoadA((props) => {

    const { actions: { venueClearState } } = props;

    testMock(venueClearState, [], []);

  }));

  it("should call initVenueData conditionally on update", () => testLoadB((props, userData) => {

    const { actions: { rsvpGetList, venueGetItem } } = props;

    const { lib: { getLocation } } = VenuePage.injected;

    const [innerA, innerB] = Object.values(userData).map((e) => [e]);

    const [outerA, outerB, outerC] = [
      ["id-a", null],
      ["id-b", null],
      ["id-b", getGeoPoint(0)]
    ];

    testMock(rsvpGetList, [], [], [], [], []);
    testMock(getLocation, innerA, innerA, innerA, innerA, innerB);
    testMock(venueGetItem, outerA, outerA, outerA, outerB, outerC);

  }));

});

describe("VenuePage (list)", () => {

  const dataList = [{ venues: { data: [{ id: "id-a" }] } }, { id: "id-a" }];

  const testSnapshot = withDataList(initTestSnapshot(testShallow), dataList);

  const testPage = (data) => testSnapshot({ venues: { data: [data] } });

  it("should match snapshot (favorites)", () => testPage({ favorites: [{}] }));

  it("should match snapshot (rsvps)", () => testPage({ rsvps: [{}] }));

});
