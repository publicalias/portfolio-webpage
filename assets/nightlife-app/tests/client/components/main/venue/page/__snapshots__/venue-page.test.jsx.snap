// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`venue page should match snapshot (authenticated) 1`] = `
<div
  className="c-venue-page"
>
  <div
    className="c-venue-page__info"
  >
    <VenueInfo
      actions={
        Object {
          "favoriteAdd": [MockFunction],
          "favoriteRemove": [MockFunction],
          "friendAdd": [MockFunction],
          "friendConfirm": [MockFunction],
          "friendDismiss": [MockFunction],
          "friendGetList": [MockFunction],
          "friendRemove": [MockFunction],
          "metaAddErrors": [MockFunction],
          "metaCloseError": [MockFunction],
          "metaGetUser": [MockFunction],
          "metaNoOp": [MockFunction],
          "metaSaveAddress": [MockFunction],
          "metaSaveAvatar": [MockFunction],
          "metaSetAddress": [MockFunction],
          "metaSetAvatar": [MockFunction],
          "metaSetLoading": [MockFunction],
          "metaSetState": [MockFunction],
          "metaTimeoutError": [MockFunction],
          "metaToggleDelete": [MockFunction],
          "metaToggleLoaded": [MockFunction],
          "metaToggleSettings": [MockFunction],
          "rsvpAdd": [MockFunction],
          "rsvpDismiss": [MockFunction],
          "rsvpGetList": [MockFunction],
          "rsvpRemove": [MockFunction],
          "userClearState": [MockFunction],
          "userGetItem": [MockFunction],
          "userGetList": [MockFunction],
          "userSetSearch": [MockFunction],
          "userToggleBlock": [MockFunction],
          "userToggleRange": [MockFunction],
          "venueClearForm": [MockFunction],
          "venueClearState": [MockFunction],
          "venueGetItem": [MockFunction],
          "venueGetList": [MockFunction],
          "venueSetMessage": [MockFunction],
          "venueSetPause": [MockFunction],
          "venueSetPhoto": [MockFunction],
          "venueSetSearch": [MockFunction],
          "venueSetStart": [MockFunction],
          "venueSetTime": [MockFunction],
          "venueToggleForm": [MockFunction],
          "venueToggleRange": [MockFunction],
          "venueToggleSort": [MockFunction],
        }
      }
      data={
        Object {
          "account": Object {
            "address": "",
            "avatar": "",
            "delete": false,
            "loaded": false,
            "settings": false,
          },
          "errors": Array [],
          "loading": false,
          "notifications": Object {
            "friends": Array [],
            "rsvps": Array [],
          },
          "user": Object {
            "auth": Object {
              "id": "",
              "provider": "",
            },
            "data": Object {
              "address": "",
              "app": "nightlife-app",
              "avatar": "",
              "blocks": Array [],
              "distance": 0,
              "favorites": Array [],
              "friends": Array [],
              "location": null,
              "restricted": false,
            },
            "id": "",
            "name": "",
            "type": "auth",
          },
          "users": Object {
            "data": Array [],
            "list": Object {
              "range": false,
              "search": "",
            },
          },
          "venues": Object {
            "data": Array [],
            "list": Object {
              "range": false,
              "search": "",
              "sort": false,
            },
            "page": Object {
              "form": Object {
                "message": "",
                "open": false,
                "time": "",
              },
              "photos": Object {
                "pause": false,
                "photo": null,
                "start": false,
              },
            },
          },
        }
      }
      history={
        Object {
          "push": [MockFunction],
        }
      }
      local={
        Object {
          "venue": Object {
            "coordinates": Object {
              "latitude": 0,
              "longitude": 0,
            },
            "display_phone": "",
            "distance": 0,
            "favorites": Array [],
            "hours": Array [],
            "id": "",
            "image_url": "",
            "location": Object {
              "display_address": Array [],
            },
            "name": "",
            "photos": Array [],
            "rating": 0,
            "rsvps": Array [],
            "url": "",
          },
        }
      }
      location={Object {}}
      match={Object {}}
      staticContext={Object {}}
    />
  </div>
  <div
    className="c-venue-page__controls"
  >
    <VenueControls
      actions={
        Object {
          "favoriteAdd": [MockFunction],
          "favoriteRemove": [MockFunction],
          "friendAdd": [MockFunction],
          "friendConfirm": [MockFunction],
          "friendDismiss": [MockFunction],
          "friendGetList": [MockFunction],
          "friendRemove": [MockFunction],
          "metaAddErrors": [MockFunction],
          "metaCloseError": [MockFunction],
          "metaGetUser": [MockFunction],
          "metaNoOp": [MockFunction],
          "metaSaveAddress": [MockFunction],
          "metaSaveAvatar": [MockFunction],
          "metaSetAddress": [MockFunction],
          "metaSetAvatar": [MockFunction],
          "metaSetLoading": [MockFunction],
          "metaSetState": [MockFunction],
          "metaTimeoutError": [MockFunction],
          "metaToggleDelete": [MockFunction],
          "metaToggleLoaded": [MockFunction],
          "metaToggleSettings": [MockFunction],
          "rsvpAdd": [MockFunction],
          "rsvpDismiss": [MockFunction],
          "rsvpGetList": [MockFunction],
          "rsvpRemove": [MockFunction],
          "userClearState": [MockFunction],
          "userGetItem": [MockFunction],
          "userGetList": [MockFunction],
          "userSetSearch": [MockFunction],
          "userToggleBlock": [MockFunction],
          "userToggleRange": [MockFunction],
          "venueClearForm": [MockFunction],
          "venueClearState": [MockFunction],
          "venueGetItem": [MockFunction],
          "venueGetList": [MockFunction],
          "venueSetMessage": [MockFunction],
          "venueSetPause": [MockFunction],
          "venueSetPhoto": [MockFunction],
          "venueSetSearch": [MockFunction],
          "venueSetStart": [MockFunction],
          "venueSetTime": [MockFunction],
          "venueToggleForm": [MockFunction],
          "venueToggleRange": [MockFunction],
          "venueToggleSort": [MockFunction],
        }
      }
      data={
        Object {
          "account": Object {
            "address": "",
            "avatar": "",
            "delete": false,
            "loaded": false,
            "settings": false,
          },
          "errors": Array [],
          "loading": false,
          "notifications": Object {
            "friends": Array [],
            "rsvps": Array [],
          },
          "user": Object {
            "auth": Object {
              "id": "",
              "provider": "",
            },
            "data": Object {
              "address": "",
              "app": "nightlife-app",
              "avatar": "",
              "blocks": Array [],
              "distance": 0,
              "favorites": Array [],
              "friends": Array [],
              "location": null,
              "restricted": false,
            },
            "id": "",
            "name": "",
            "type": "auth",
          },
          "users": Object {
            "data": Array [],
            "list": Object {
              "range": false,
              "search": "",
            },
          },
          "venues": Object {
            "data": Array [],
            "list": Object {
              "range": false,
              "search": "",
              "sort": false,
            },
            "page": Object {
              "form": Object {
                "message": "",
                "open": false,
                "time": "",
              },
              "photos": Object {
                "pause": false,
                "photo": null,
                "start": false,
              },
            },
          },
        }
      }
      history={
        Object {
          "push": [MockFunction],
        }
      }
      local={
        Object {
          "venue": Object {
            "coordinates": Object {
              "latitude": 0,
              "longitude": 0,
            },
            "display_phone": "",
            "distance": 0,
            "favorites": Array [],
            "hours": Array [],
            "id": "",
            "image_url": "",
            "location": Object {
              "display_address": Array [],
            },
            "name": "",
            "photos": Array [],
            "rating": 0,
            "rsvps": Array [],
            "url": "",
          },
        }
      }
      location={Object {}}
      match={Object {}}
      staticContext={Object {}}
    />
  </div>
  <div
    className="c-venue-page__favorites"
  >
    <VenueList
      local={
        Object {
          "heading": "Favorites",
          "list": Array [],
        }
      }
    />
  </div>
  <div
    className="c-venue-page__rsvps"
  >
    <VenueList
      local={
        Object {
          "heading": "RSVPs",
          "list": Array [],
        }
      }
    />
  </div>
</div>
`;

exports[`venue page should match snapshot (default) 1`] = `
<div
  className="c-venue-page"
>
  <div
    className="c-venue-page__info"
  >
    <VenueInfo
      actions={
        Object {
          "favoriteAdd": [MockFunction],
          "favoriteRemove": [MockFunction],
          "friendAdd": [MockFunction],
          "friendConfirm": [MockFunction],
          "friendDismiss": [MockFunction],
          "friendGetList": [MockFunction],
          "friendRemove": [MockFunction],
          "metaAddErrors": [MockFunction],
          "metaCloseError": [MockFunction],
          "metaGetUser": [MockFunction],
          "metaNoOp": [MockFunction],
          "metaSaveAddress": [MockFunction],
          "metaSaveAvatar": [MockFunction],
          "metaSetAddress": [MockFunction],
          "metaSetAvatar": [MockFunction],
          "metaSetLoading": [MockFunction],
          "metaSetState": [MockFunction],
          "metaTimeoutError": [MockFunction],
          "metaToggleDelete": [MockFunction],
          "metaToggleLoaded": [MockFunction],
          "metaToggleSettings": [MockFunction],
          "rsvpAdd": [MockFunction],
          "rsvpDismiss": [MockFunction],
          "rsvpGetList": [MockFunction],
          "rsvpRemove": [MockFunction],
          "userClearState": [MockFunction],
          "userGetItem": [MockFunction],
          "userGetList": [MockFunction],
          "userSetSearch": [MockFunction],
          "userToggleBlock": [MockFunction],
          "userToggleRange": [MockFunction],
          "venueClearForm": [MockFunction],
          "venueClearState": [MockFunction],
          "venueGetItem": [MockFunction],
          "venueGetList": [MockFunction],
          "venueSetMessage": [MockFunction],
          "venueSetPause": [MockFunction],
          "venueSetPhoto": [MockFunction],
          "venueSetSearch": [MockFunction],
          "venueSetStart": [MockFunction],
          "venueSetTime": [MockFunction],
          "venueToggleForm": [MockFunction],
          "venueToggleRange": [MockFunction],
          "venueToggleSort": [MockFunction],
        }
      }
      data={
        Object {
          "account": Object {
            "address": "",
            "avatar": "",
            "delete": false,
            "loaded": false,
            "settings": false,
          },
          "errors": Array [],
          "loading": false,
          "notifications": Object {
            "friends": Array [],
            "rsvps": Array [],
          },
          "user": Object {},
          "users": Object {
            "data": Array [],
            "list": Object {
              "range": false,
              "search": "",
            },
          },
          "venues": Object {
            "data": Array [],
            "list": Object {
              "range": false,
              "search": "",
              "sort": false,
            },
            "page": Object {
              "form": Object {
                "message": "",
                "open": false,
                "time": "",
              },
              "photos": Object {
                "pause": false,
                "photo": null,
                "start": false,
              },
            },
          },
        }
      }
      history={
        Object {
          "push": [MockFunction],
        }
      }
      local={
        Object {
          "venue": Object {
            "coordinates": Object {
              "latitude": 0,
              "longitude": 0,
            },
            "display_phone": "",
            "distance": 0,
            "favorites": Array [],
            "hours": Array [],
            "id": "",
            "image_url": "",
            "location": Object {
              "display_address": Array [],
            },
            "name": "",
            "photos": Array [],
            "rating": 0,
            "rsvps": Array [],
            "url": "",
          },
        }
      }
      location={Object {}}
      match={Object {}}
      staticContext={Object {}}
    />
  </div>
  <div
    className="c-venue-page__favorites"
  >
    <VenueList
      local={
        Object {
          "heading": "Favorites",
          "list": Array [],
        }
      }
    />
  </div>
  <div
    className="c-venue-page__rsvps"
  >
    <VenueList
      local={
        Object {
          "heading": "RSVPs",
          "list": Array [],
        }
      }
    />
  </div>
</div>
`;

exports[`venue page should match snapshot (match) 1`] = `
<div
  className="c-venue-page"
>
  <div
    className="c-venue-page__info"
  >
    <VenueInfo
      actions={
        Object {
          "favoriteAdd": [MockFunction],
          "favoriteRemove": [MockFunction],
          "friendAdd": [MockFunction],
          "friendConfirm": [MockFunction],
          "friendDismiss": [MockFunction],
          "friendGetList": [MockFunction],
          "friendRemove": [MockFunction],
          "metaAddErrors": [MockFunction],
          "metaCloseError": [MockFunction],
          "metaGetUser": [MockFunction],
          "metaNoOp": [MockFunction],
          "metaSaveAddress": [MockFunction],
          "metaSaveAvatar": [MockFunction],
          "metaSetAddress": [MockFunction],
          "metaSetAvatar": [MockFunction],
          "metaSetLoading": [MockFunction],
          "metaSetState": [MockFunction],
          "metaTimeoutError": [MockFunction],
          "metaToggleDelete": [MockFunction],
          "metaToggleLoaded": [MockFunction],
          "metaToggleSettings": [MockFunction],
          "rsvpAdd": [MockFunction],
          "rsvpDismiss": [MockFunction],
          "rsvpGetList": [MockFunction],
          "rsvpRemove": [MockFunction],
          "userClearState": [MockFunction],
          "userGetItem": [MockFunction],
          "userGetList": [MockFunction],
          "userSetSearch": [MockFunction],
          "userToggleBlock": [MockFunction],
          "userToggleRange": [MockFunction],
          "venueClearForm": [MockFunction],
          "venueClearState": [MockFunction],
          "venueGetItem": [MockFunction],
          "venueGetList": [MockFunction],
          "venueSetMessage": [MockFunction],
          "venueSetPause": [MockFunction],
          "venueSetPhoto": [MockFunction],
          "venueSetSearch": [MockFunction],
          "venueSetStart": [MockFunction],
          "venueSetTime": [MockFunction],
          "venueToggleForm": [MockFunction],
          "venueToggleRange": [MockFunction],
          "venueToggleSort": [MockFunction],
        }
      }
      data={
        Object {
          "account": Object {
            "address": "",
            "avatar": "",
            "delete": false,
            "loaded": false,
            "settings": false,
          },
          "errors": Array [],
          "loading": false,
          "notifications": Object {
            "friends": Array [],
            "rsvps": Array [],
          },
          "user": Object {},
          "users": Object {
            "data": Array [],
            "list": Object {
              "range": false,
              "search": "",
            },
          },
          "venues": Object {
            "data": Array [
              Object {
                "coordinates": Object {
                  "latitude": 0,
                  "longitude": 0,
                },
                "display_phone": "",
                "distance": 0,
                "favorites": Array [],
                "hours": Array [],
                "id": "id-a",
                "image_url": "",
                "location": Object {
                  "display_address": Array [],
                },
                "name": "",
                "photos": Array [],
                "rating": 0,
                "rsvps": Array [],
                "url": "",
              },
            ],
            "list": Object {
              "range": false,
              "search": "",
              "sort": false,
            },
            "page": Object {
              "form": Object {
                "message": "",
                "open": false,
                "time": "",
              },
              "photos": Object {
                "pause": false,
                "photo": null,
                "start": false,
              },
            },
          },
        }
      }
      history={
        Object {
          "push": [MockFunction],
        }
      }
      local={
        Object {
          "venue": Object {
            "coordinates": Object {
              "latitude": 0,
              "longitude": 0,
            },
            "display_phone": "",
            "distance": 0,
            "favorites": Array [],
            "hours": Array [],
            "id": "id-a",
            "image_url": "",
            "location": Object {
              "display_address": Array [],
            },
            "name": "",
            "photos": Array [],
            "rating": 0,
            "rsvps": Array [],
            "url": "",
          },
        }
      }
      location={Object {}}
      match={Object {}}
      staticContext={Object {}}
    />
  </div>
  <div
    className="c-venue-page__favorites"
  >
    <VenueList
      local={
        Object {
          "heading": "Favorites",
          "list": Array [],
        }
      }
    />
  </div>
  <div
    className="c-venue-page__rsvps"
  >
    <VenueList
      local={
        Object {
          "heading": "RSVPs",
          "list": Array [],
        }
      }
    />
  </div>
</div>
`;
